### PR TITLE
improve speed of PR builds + remove duplicated comments

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/1201614831475344/1208479715973664/f

- [x] re-use the `npm` cache correctly to speed up builds
- [x] update the PR comments in place

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#1090** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1090?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/build-pr.yml [hunk 1]
```diff
@@ -42,8 +47,18 @@ jobs:
           echo "BRANCH_NAME=pr-releases/pr-${PR_NUMBER}" >> $GITHUB_ENV
           echo "COMMIT_HASH=$(git rev-parse HEAD)" >> $GITHUB_ENV
 
-      - name: Post comment on PR
+      - name: Find Previous Comment
+        uses: peter-evans/find-comment@v2
```
Comment (shakyShane) [inline .github/workflows/build-pr.yml:51]: I've looked through the code in this action - it's very simple and we could implement it ourselves if this becomes an issue.

File: .github/workflows/build-pr.yml [hunk 2]
```diff
@@ -9,20 +9,25 @@ permissions: write-all
...
+        with:
+          node-version: 20.x
+      - uses: actions/cache@v4
         with:
-          node-version: '20'
-          cache: 'npm'
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
Comment (shakyShane) [inline .github/workflows/build-pr.yml:27]: this ensures the cache from other runners will be used

**Review by mgurgel — APPROVED** [review]
Excellent

**Pull request merged**